.github/workflows/ci.yml

```diff
@@ -21,6 +19,36 @@ jobs:
       - name: restore build & cargo cache
         uses: Swatinem/rust-cache@v1
 
+      - name: Build
+        run: cargo build --workspace --locked
+
+      - name: compile test binaries
+        run: |
+          cargo test --no-run
+          mkdir ./test-binaries/
+          find ./target \
+            -name "docs_rs*" \
+            -executable -type f \
+            -exec cp {} ./test-binaries/ \;
+
+      - name: set up test binaries to cache
```

Comment:
Rather than using caches for this we should use artifacts (with an expiration of 1 day).

The problem with using caches is that there is a cap for how many caches we can store in the repository, and if we go over that limit older caches will be removed. Artifacts use a different storage quota instead.

Reply:
I added a commit doing this. 

It needed a small `chmod +x` for the restore  [since the artifacts don't copy permissions](https://github.com/actions/upload-artifact#maintaining-file-permissions-and-case-sensitive-files).

Reply:
Great!